python/spacetime/store.py

```diff
@@ -376,8 +376,22 @@ def register_app(self, app, typemap, name2class, name2baseclasses):
             set(typemap.setdefault("gettingsetting", set())),
             set(typemap.setdefault("setting", set()))
           )
+        if app in self.__copylock:
+            try:
+                with self.__copylock[app]:
+                    # Clean-up old registration
+                    for strtp in self.__type_to_app.keys():
+                        if app in self.__type_to_app[strtp]:
+                            del self.__type_to_app[strtp][app]
+
+                    self.__app_to_basechanges[app] = {}
+                    del self.__copylock[app]
+            except:
+                pass
+
```

Comment:
we should load the "new" part of app_to_basechanges with the objects of all the types it asks for at the end of this function.